### DISCUSSION TITLE
Any step by step how to documentation on synthesizing with a pre trained model

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#92] [Archived]
<USER>>>> robertgass<br/> <DATE> [September 19, 2019, 1:39pm]<br/> <br/> I've looked for the last couple hours, but haven't had much success.<br/> I've tried running synthesize.py with config_tacotron2.json and a<br/> downloaded release model, but it crashed and burned with 'Error(s) in<br/> loading state_dict'.<br/> <br/> I'm literally taking guesses as to how to run this. What documentation<br/> have I missed??<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/any-step-by-step-how-to-documentation-on-synthesizing-with-a-pre-trained-model]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> anotherdirtbag<br/> <DATE> [September 19, 2019, 6:18pm]<br/> <br/> I've only gotten it to work from a pretrained wavernn+tacotron2 per the<br/> links here: <https://github.com/erogol/WaveRNN> *slash* <br/> for mozilla tts i had to 'git checkout 8a47b46' *slash* <br/> and for wavernn i used 'git checkout 8a1c152' *slash* <br/> Then I followed one of the older 'notebooks' on github and came up with<br/> this. *slash* <br/> <https://gist.github.com/anotherdirtbag/a03a8826ddf04bed0f1433bbc84032b0><br/> <br/> There's a few issues in the model I've got working though. Namely, the<br/> beginning second of each sentence is cutoff, there's some popping noise,<br/> and unpredictable silence in some cases.<br/> <br/> Edit: Frankly, this guy's work sounds better than mine and he didn't<br/> even use WaveRNN. [My progress on expressive speech<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> geneing<br/> <DATE> [September 23, 2019, 5:59am]<br/> <br/> A few pointers when it comes to using both pretrained models or when<br/> training your own. I probably have made all these mistakes a few times.<br/> <br/> 1. Make sure that WaveRNN and TTS are configured with the same sampling<br/> rate.<br/> 2. Make sure that the following parameters match between the two:<br/> symmetric_norm, signal_norm, max_norm, clip_norm. In general, always<br/> compare configuration files for WaveRNN and TTS to make sure they<br/> are compatible.<br/> 3. For WaveRNN, make sure that upsample_factors match the hop length.<br/> For example, for 16KHz sampling rate and frame_shift_ms = 12.5, the<br/> hop length is 200 frames. Product of upsample factors 5*5*8=200,<br/> which is correct.<br/> 4. Silence trimming should be on if dataset has some silence in the<br/> beginning.<br/> 5. Both wavernn and tactron pretrained models often have config files<br/> that are incompatible with current source code (and I think so are<br/> some config files checked in).<br/> <br/> For WaveRNN training - subjective quality continues to improve even as<br/> loss function is not decreasing. I usually need about 100K steps with<br/> 10bit output before the speech sounds 'good'. Mixture of logistics<br/> probably needs even more steps.<br/> <br/> Cutoff sentences and unpredictable silences are probably symptoms of<br/> attention not working well, especially if it happens in longer<br/> sentences. Attention is always tricky in tacotron models. Current TTS<br/> dev branch has scheduled training, which works really well for tacotron.<br/> <br/> [Archived Post]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> anotherdirtbag<br/> <DATE> [September 25, 2019, 3:33pm]<br/> <br/> Thank you. that's really helpful info. I'm just starting into this and<br/> was unsuccessful in my first and only training attempt (thus finding<br/> pretrained models). I'll try training again with newer source code and<br/> see if I can figure it out.<br/> <br/> [Archived Post]<br/>
